packages/server-renderer/src/index.ts

```diff
@@ -12,21 +13,147 @@ export interface ServerRequest {
 
 export interface ServerRendererV1 {
   readonly serverRequest: ServerRequest | undefined;
+
+  startRendering(render: () => string): Promise<string>;
+  waitForFeatureApp(
+    featureAppClientUrl: string,
+    featureAppLoadingEvent: Promise<void>
+  ): void;
+  register(isCompleted: () => boolean): void;
+  rerender(): Promise<void>;
 }
 
 export interface SharedServerRenderer extends SharedFeatureService {
   readonly '1.0': FeatureServiceBinder<ServerRendererV1>;
 }
 
+interface PromiseWithStatus<TResult> {
+  readonly promise: Promise<TResult>;
+  status: 'new' | 'pending' | 'final';
```

Comment:
Is there maybe a more elegant approach to track whether we handled feature app module promises, and whether they are resolved/rejected?